### PR TITLE
Product bundles in Order Creation

### LINKED ISSUE #9541: Product Bundles support during order creation
Adding product bundles' support during order creation will provide a consistent experience for merchants, and it could increase the app’s revenue and engagement.

> Need to support variable items, composite items, and bundled items. I can add these to the order via the website and the order shows up correctly in the app, but I can’t add them to the order via the app because they show up as $0.00 items.

- [ ] Display the configuration button https://github.com/woocommerce/woocommerce-android/pull/9646
- [ ] Change configuration text to Edit when bundled is configured
- [ ] Add a new bundle using the app
- [ ] Edit a new bundle using the app
- [ ] Remove a bundle from the app

### PR DESCRIPTION
**Pull request by atorresveiga**
Closes: #9541

### Why
As part of the support for extensions milestone 2, we are working on support bundle products in the order creation flow.

### Description
This PR is the feature branch for product bundles in the order creation flow

#### Prerequisites 

1.  Install the product bundles extension
2. Create a bundle product requiring a configuration
3. Create a bundle product that doesn't require a configuration*
4. Create a product that is not a bundle product.

(*) A simple bundle product will need a configuration when it is optional or has min-max rules. If min = max, we should use this value as the product quantity, meaning this bundled product doesn't require quantity configuration. Variable products are out of the scope of this PR.

#### Testing
You can follow the testing plan here pe5pgL-3Ze-p2

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/59913df2-e459-4fd6-835b-cbb986a8d66c

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/bundle/OrderItemRules.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package com.woocommerce.android.ui.orders.creation.bundle
+
+class OrderItemRules private constructor(
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/bundle/OrderItemRules.kt:3]: This class will be refactored on coming PRs while we add more rules to the item configurations.

**Comment by ThomazFB** [comment]
@atorresveiga I've been following this PR and I noticed that you marked as ready for review but the `do not merge` label is still on. Should Is tart reviewing this PR?

**Comment by atorresveiga** [comment]
Sorry for the late answer Thomaz! You can review it, but my plan is to merge all bundles PR into this one and from here merge the code to trunk, like a feature branch. There are a lot of changes that I want to deeply test first before releasing this feature. This PR is an intent of not to create a huge PR

**Review by 0nko — APPROVED** [review]
:shipit:

**Pull request merged**